lib/kafka/protocol/message.rb

```diff
@@ -71,14 +72,15 @@ def self.decode(decoder)
         end
 
         attributes = message_decoder.int8
+        timestamp = message_decoder.int64
         key = message_decoder.bytes
         value = message_decoder.bytes
 
         # The codec id is encoded in the three least significant bits of the
         # attributes.
         codec_id = attributes & 0b111
 
-        new(key: key, value: value, codec_id: codec_id, offset: offset)
+        new(key: key, value: value, codec_id: codec_id, offset: offset, create_time: Time.at(timestamp/1000))
```

Comment:
This should divide by a float in order to preserve milliseconds (ie. `1000.0`)

For example:
```
$ irb
$ timestamp = 1473867722801 # milliseconds
 => 1473867722801
$ Time.at(timestamp/1000).strftime('%S.%N')
 => "02.000000000"
$ Time.at(timestamp/1000.0).strftime('%S.%N')
 => "02.801000118"
```